dev-docs/RFCs/v8.0/ui-module-rfc.md

```diff
@@ -0,0 +1,137 @@
+# UI Module RFC
+
+* **Author**: Xiaoji Chen
+* **Date**: December 2022
+* **Status**: **Pending Review**
+
+## Overview
+
+Visualization applications often require some chrome around the WebGL canvas to offer better user experience. While it is possible for every app to implement their own UI components that interact with deck.gl, it is no trivial effort to implement them well. 
+
+This RFC proposes to add a new module that packages common UI components that seamlessly integrate with deck.gl, including but not limited to:
```

Comment:
Other use cases:

Animation with timeline libs, such as [TheatreJS](https://www.theatrejs.com/) or [motion dev tools](https://motion.dev/tools).

Layer designer panels to speed up visual iteration using libraries, such as dat.gui or [Leva](https://github.com/pmndrs/leva).

In-world control widgets for positioning views, lighting, and other effects. Similar to [transform controls](https://threejs.org/examples/?q=controls#misc_controls_transform) and [arc balls](https://threejs.org/examples/?q=controls#misc_controls_arcball) in three.js

nebula.gl (or libraries like it) could utilize this for their editable map UI elements.

Invert the implementation of @deck.gl/mapbox and @deck.gl/google-maps, so that we can specify a unified, neutral API for basemaps to implement.